src/hypothesis/strategies.py

```diff
@@ -711,6 +711,29 @@ def text(
 
 @cacheable
 @defines_strategy
+def text_matching(regex):
```

Comment:
Without bikeshedding, I think there should be more discussion - [giving useful names is hard](http://thecodelesscode.com/case/220) but also important and it's much harder to change later.